src/EventSubscriber/SetCookieSubscriber.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class SetCookieSubscriber implements EventSubscriberInterface
+{
+    public function onKernelResponse(ResponseEvent $event)
+    {
+        if (
+            !$event->isMainRequest() ||
+            null === $cookie = $event->getRequest()->attributes->get('_mercure_authorization_cookie')) {
+            return;
+        }
+
```

Comment:
You could remove the cookie from the request I think

Reply:
Good idea, and thanks for the explanations in DM!!